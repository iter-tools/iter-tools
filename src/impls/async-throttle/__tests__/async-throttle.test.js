import { asyncThrottle } from 'iter-tools-es';
import { asyncWrap, asyncUnwrap, anyType } from '../../../test/async-helpers.js';

describe('asyncThrottle', () => {
  it('throttle the output', async () => {
    const iter = asyncThrottle(10, asyncWrap([0, 1, 2, 3, 4, 5]));
    const t0 = Date.now();
    expect(await asyncUnwrap(iter)).toEqual([0, 1, 2, 3, 4, 5]);
    const t1 = Date.now();
    expect(t1 - t0).toBeGreaterThanOrEqual(40);
  });

  it('returns empty output when passed null', async () => {
    expect(await asyncUnwrap(asyncThrottle(10, null))).toEqual([]);
  });

  describe('when interval is invalid', () => {
    it('throws', async () => {
      expect(() => asyncThrottle(0, null)).toThrowErrorMatchingSnapshot();
      expect(() => asyncThrottle(anyType(null), null)).toThrowErrorMatchingSnapshot();
    });
  });
});
