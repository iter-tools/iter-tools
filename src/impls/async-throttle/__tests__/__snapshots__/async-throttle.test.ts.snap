// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`asyncThrottle when interval is invalid throws 1`] = `"intervalMs argument to asyncThrottle must be a number > 0"`;

exports[`asyncThrottle when interval is invalid throws 2`] = `"intervalMs argument to asyncThrottle must be a number > 0"`;
